cnf-certification-test/accesscontrol/securitycontextcontainer/securitycontextcontainer.go

```diff
@@ -0,0 +1,454 @@
+	FsGroupPresent             OkNok
+	SeLinuxContextPresent      OkNok
+	Capabilities               string
+	HaveDropCapabilities       OkNok
+	AllVolumeAllowed           OkNok
+}
+
+var (
+	requiredDropCapabilities = []string{"MKNOD", "SETUID", "SETGID", "KILL"}
+	dropAll                  = []string{"ALL"}
+	category2AddCapabilities = []string{"NET_ADMIN, NET_RAW"}
+	category3AddCapabilities = []string{"NET_ADMIN, NET_RAW, IPC_LOCK"}
+	Category1                = ContainerSCC{NOK,
+		NOK,
+		NOK,
+		NOK,
+		NOK,
+		OK,
+		NOK,
+		OK,
+		NOK,
+		OK,
+		OK,
+		OK,
+		category1,
+		OK,
+		OK}
+
+	Category1NoUID0 = ContainerSCC{NOK,
+		NOK,
+		NOK,
+		NOK,
+		NOK,
+		OK,
+		NOK,
+		NOK,
+		NOK,
+		NOK,
+		OK,
+		OK,
+		category1,
+		OK,
+		OK}
+
+	Category2 = ContainerSCC{NOK,
+		NOK,
+		NOK,
+		NOK,
+		NOK,
+		OK,
+		NOK,
+		OK,
+		NOK,
+		OK,
+		OK,
+		OK,
+		"category2",
+		OK,
+		OK}
+
+	Category3 = ContainerSCC{NOK,
+		NOK,
+		NOK,
+		NOK,
+		NOK,
+		OK,
+		NOK,
+		OK,
+		NOK,
+		OK,
+		OK,
+		OK,
+		"category3",
+		OK,
+		OK}
+)
+
+//nolint:gocritic,gocyclo
+func GetContainerSCC(cut *provider.Container, containerSCC ContainerSCC) ContainerSCC {
+	containerSCC.HostPorts = NOK
+	for _, aPort := range cut.Ports {
+		if aPort.HostPort != 0 {
+			containerSCC.HostPorts = OK
+			break
+		}
+	}
+	updateCapabilities(cut, &containerSCC)
+	containerSCC.PrivilegeEscalation = NOK
+	if cut.SecurityContext != nil && cut.SecurityContext.AllowPrivilegeEscalation != nil {
+		containerSCC.PrivilegeEscalation = OK
+	}
+	containerSCC.PrivilegedContainer = NOK
+	if cut.SecurityContext != nil && cut.SecurityContext.Privileged != nil && *(cut.SecurityContext.Privileged) {
+		containerSCC.PrivilegedContainer = OK
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.RunAsUser != nil {
+		containerSCC.RunAsUserPresent = OK
+	}
+	containerSCC.ReadOnlyRootFilesystem = NOK
+	if cut.SecurityContext != nil && cut.SecurityContext.ReadOnlyRootFilesystem != nil && *cut.SecurityContext.ReadOnlyRootFilesystem {
+		containerSCC.ReadOnlyRootFilesystem = OK
+	}
+	containerSCC.RunAsNonRoot = NOK
+	if cut.SecurityContext != nil && cut.SecurityContext.RunAsNonRoot != nil && *cut.SecurityContext.RunAsNonRoot {
+		containerSCC.RunAsNonRoot = OK
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.SELinuxOptions != nil {
+		containerSCC.SeLinuxContextPresent = OK
+	}
+	return containerSCC
+}
+
+func updateCapabilities(cut *provider.Container, containerSCC *ContainerSCC) {
+	containerSCC.HaveDropCapabilities = NOK
+	if cut.SecurityContext != nil && cut.SecurityContext.Capabilities != nil {
+		var sliceDropCapabilities []string
+		for _, ncc := range cut.SecurityContext.Capabilities.Drop {
+			sliceDropCapabilities = append(sliceDropCapabilities, string(ncc))
+		}
+		sort.Strings(sliceDropCapabilities)
+
+		sort.Strings(requiredDropCapabilities)
+		if reflect.DeepEqual(sliceDropCapabilities, requiredDropCapabilities) || reflect.DeepEqual(sliceDropCapabilities, dropAll) {
```

Comment:
Thinking more about it, here equal might be too restrictive. If a container has more DropCapabilities than needed for all categories it should not fail this test. So instead of equal we should test that the container DropCapabilities list contains at least the required ones